tests/test_util.py

```diff
@@ -41,3 +62,177 @@ class A:
+
+def test_sshmanager_open(sshmanager_fix):
+    con = sshmanager_fix.open("localhost")
+    assert isinstance(con, SSHConnection)
+
+def test_sshmanager_add_forward(sshmanager_fix):
+    port = sshmanager_fix.request_forward("localhost", 'localhost', 3000)
+    assert port < 65536
+
+def test_sshmanager_remove_forward(sshmanager_fix):
+    port = sshmanager_fix.request_forward("localhost", 'localhost', 3000)
+    sshmanager_fix.remove_forward('localhost', 'localhost', 3000)
+    assert 3000 not in sshmanager_fix.get('localhost')._forwards
+
+def test_sshmanager_close(sshmanager_fix):
+    con = sshmanager_fix.open("localhost")
+    assert isinstance(con, SSHConnection)
+    sshmanager_fix.close("localhost")
+
+def test_sshmanager_remove_raise(sshmanager_fix):
+    con = sshmanager_fix.open("localhost")
+    con.connect()
+    with pytest.raises(ExecutionError):
+        sshmanager_fix.remove_connection(con)
+
+def test_sshmanager_add_duplicate(sshmanager_fix):
+    host = 'localhost'
+    con = SSHConnection(host)
+    sshmanager_fix.add_connection(con)
+    con_there = sshmanager_fix._connections[host]
+    sshmanager_fix.add_connection(con)
+    con_now = sshmanager_fix._connections[host]
+    assert con_now == con_there
+
+def test_sshmanager_add_new(sshmanager_fix):
+    host = 'other_host'
+    con = SSHConnection(host)
+    sshmanager_fix.add_connection(con)
+    con_now = sshmanager_fix._connections[host]
+    assert con_now == con
+
+
+def test_sshmanager_remove_raise():
+    con = SSHConnection("nosuchhost.notavailable")
+    with pytest.raises(ExecutionError):
+        con.connect()
+
+def test_proxymanager_no_proxy(target):
+    host = 'localhost'
+    port = 5000
+    nr = NetworkSerialPort(target, host=host, port=port, name=None)
+    nr.proxy = ''
+    nr.proxy_required = False
+    assert (host, port) == proxymanager.get_host_and_port(nr)
+
+def test_proxymanager_remote_forced_proxy(target):
+    sshmanager.close_all()
+    host= 'localhost'
+    port = 5000
+    nr = NetworkSerialPort(target, host=host, port=port, name=None)
+    nr.proxy = 'localhost'
+    nr.proxy_required = True
+    nhost, nport = proxymanager.get_host_and_port(nr)
+    assert (host, port) != (nhost, nport)
+    sshmanager.close_all()
+
+def test_proxymanager_local_forced_proxy(target):
+    host= 'localhost'
+    port = 5000
+    nr = NetworkSerialPort(target, host=host, port=port, name=None)
+    nr.proxy = 'localhost'
+    nr.proxy_required = True
+    nhost, nport = proxymanager.get_host_and_port(nr)
+    assert (host, port) != (nhost, nport)
+
+def test_remote_managedfile(target, tmpdir):
+    import hashlib
+    import getpass
+
+    res = NetworkResource(target, "test", "localhost")
+    t = tmpdir.join("test")
+    t.write(
+"""
+Test
+"""
+    )
+    hash = hashlib.sha256(t.read().encode("utf-8")).hexdigest()
+    mf = ManagedFile(t,res)
+    mf.sync_to_resource()
+    assert os.path.isfile("/tmp/labgrid/{}/{}/test".format(getpass.getuser(), hash))
```

Comment:
Using getpass will import other modules (such as termios).

Basically we just need (in util.py):
```python
def get_user():
    user = os.environ.get("USER")
    if user:
        return user
    import pwd
    return pwd.getpwuid(os.getuid())[0]
```

Reply:
I can add this in a separate commit.